ui/lesson/TerminalChallenge.tsx

```diff
@@ -62,8 +62,25 @@ export default function TerminalChallenge({
           },
         ]
   )
+  const [windowWidth, setWindowWidth] = useState(window.innerWidth)
 
-  const isSmallScreen = useMediaQuery({ query: '(max-width: 767px)' })
+  useEffect(() => {
```

Comment:
I think I prefer keeping the `useMediaQuery` hook rather than moving the resizing logic into a specific component like `TerminalChallenge`. If, in the future, we have to do anything related to resizing the window, we would have to reimplement the same logic again using your approach. However, if it were just a hook, we could simply import and use it.

Reply:
I agree we should implement it as a reusable hook. The current `useMediaQuery` hook is not working as we intend it to work, I will try to make a new implementation for it.